genesis/sensors/contact_force.py

```diff
@@ -299,7 +343,25 @@ def _update_shared_cache(
         cls._add_noise_drift_bias(shared_metadata, shared_cache)
         shared_cache_per_sensor = shared_cache.reshape(shared_cache.shape[0], -1, 3)  # B, n_sensors * 3
         # clip for max force
-        shared_cache_per_sensor.clamp_(max=shared_metadata.max_force)
+        shared_cache_per_sensor.clamp_(min=-shared_metadata.max_force, max=shared_metadata.max_force)
         # set to 0 for undetectable force
         shared_cache_per_sensor[torch.abs(shared_cache_per_sensor) < shared_metadata.min_force] = 0.0
         cls._quantize_to_resolution(shared_metadata.resolution, shared_cache)
+
+    def _draw_debug(self, context: "RasterizerContext"):
+        """
+        Draw debug arrow representing the contact force.
+
+        Only draws for first environment.
+        """
+        envs_idx = 0 if self._manager._sim.n_envs > 0 else None
```

Comment:
You should not do this. You should rendering for the selected rendering env, ie
```python
env_idx = context.rendered_envs_idx[0]
```

Note that it should be possible to add batching support to all `draw_debug_*` methods, but let's ignore this for now.